daprovider/das/data_streaming/receiver.go

```diff
@@ -176,7 +176,13 @@ func (ms *messageStore) addNewChunk(id MessageId, chunkId uint64, chunk []byte)
 	// Validate chunk size
 	chunkLen := uint64(len(chunk))
 	if chunkId+1 == uint64(len(message.chunks)) {
-		expectedLen := message.expectedTotalSize % message.expectedChunkSize
+		// For the last chunk, if totalSize is an exact multiple of chunkSize,
+		// the expected size is chunkSize (not zero remainder).
+		rem := message.expectedTotalSize % message.expectedChunkSize
+		expectedLen := message.expectedChunkSize
+		if rem != 0 {
+			expectedLen = rem
+		}
```

Comment:
or more briefly:
```suggestion
        expectedLen := ((message.expectedTotalSize - 1) % message.expectedChunkSize) + 1
```

+ an assertion in the `startStream` that the `message.expectedTotalSize` is not `0` (which actually is also a small bug)